### ISSUE TITLE
setting creds doesnt work

### ISSUE THREAD

**Issue by chrissxYT** [Amplissimus/Amplissimus#3]
**Describe the bug**
the console says: validate() was called on null and the dialog to change creds doesnt close

**To Reproduce**
1. manually change your shared prefs to only have the password and not the username (and first_login=false)
2. open the change login data box
3. enter username
4. click save
5. see bug

**Expected behavior**
it should save and close the popup

**Stacktrace**
a stack trace should be more helpful:
```
flutter: The following NoSuchMethodError was thrown while handling a gesture:
flutter: The method 'validate' was called on null.
flutter: Receiver: null
flutter: Tried calling: validate()
flutter:
flutter: When the exception was thrown, this was the stack:
flutter: #0      Object.noSuchMethod  (dart:core-patch/object_patch.dart:51:5)
flutter: #1      _MyHomePageState.showInputEntryCredentials.<anonymous closure>.<anonymous closure> 
package:Amplissimus/main.dart:313
flutter: #2      _InkResponseState._handleTap 
package:flutter/…/material/ink_well.dart:992
flutter: #3      _InkResponseState.build.<anonymous closure> 
package:flutter/…/material/ink_well.dart:1098
flutter: #4      GestureRecognizer.invokeCallback 
package:flutter/…/gestures/recognizer.dart:184
flutter: #5      TapGestureRecognizer.handleTapUp 
package:flutter/…/gestures/tap.dart:524
flutter: #6      BaseTapGestureRecognizer._checkUp 
package:flutter/…/gestures/tap.dart:284
flutter: #7      BaseTapGestureRecognizer.handlePrimaryPointer 
package:flutter/…/gestures/tap.dart:219
flutter: #8      PrimaryPointerGestureRecognizer.handleEvent 
package:flutter/…/gestures/recognizer.dart:477
flutter: #9      PointerRouter._dispatch 
package:flutter/…/gestures/pointer_router.dart:78
flutter: #10     PointerRouter._dispatchEventToRoutes.<anonymous closure> 
package:flutter/…/gestures/pointer_router.dart:124
flutter: #11     _LinkedHashMapMixin.forEach  (dart:collection-patch/compact_hash.dart:377:8)
flutter: #12     PointerRouter._dispatchEventToRoutes 
package:flutter/…/gestures/pointer_router.dart:122
flutter: #13     PointerRouter.route 
package:flutter/…/gestures/pointer_router.dart:108
flutter: #14     GestureBinding.handleEvent 
package:flutter/…/gestures/binding.dart:220
flutter: #15     GestureBinding.dispatchEvent 
package:flutter/…/gestures/binding.dart:200
flutter: #16     GestureBinding._handlePointerEvent 
package:flutter/…/gestures/binding.dart:158
flutter: #17     GestureBinding._flushPointerEventQueue 
package:flutter/…/gestures/binding.dart:104
flutter: #18     GestureBinding._handlePointerDataPacket 
package:flutter/…/gestures/binding.dart:88
flutter: #22     _invoke1  (dart:ui/hooks.dart:283:10)
flutter: #23     _dispatchPointerDataPacket  (dart:ui/hooks.dart:192:5)
flutter: (elided 3 frames from dart:async)
flutter:
flutter: Handler: "onTap"
flutter: Recognizer:
flutter:   TapGestureRecognizer#67eb0
```

**Versions (please complete the following information):**
 - Windows 10 19something and iOS 13.5 (probably also others)
 - 1.1.3, 1.1.4

**Comment by chrissxYT** [comment]
just reproduced this on 1.1.4 on iOS 13.5 just by opening the login data box and clicking save

**Comment by chrissxYT** [comment]
this was fixed by cb42943d06a66e271c78663bc7ed546eca8ea26b
